src/test/java/org/springframework/data/elasticsearch/junit/jupiter/ClusterConnection.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.junit.jupiter;
+
+import java.io.IOException;
+import java.net.URL;
+
+import org.elasticsearch.node.Node;
+import org.elasticsearch.node.NodeValidationException;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.elasticsearch.Utils;
+import org.springframework.lang.Nullable;
+import org.springframework.util.StringUtils;
+
+/**
+ * This class manages the connection to an Elasticsearch Cluster, starting a local one if necessary. The information
+ * about the ClusterConnection is stored in a static ThreadLocal<ClusterConnectionInfo> and can be accessed with the
+ * {@link ClusterConnection#getClusterConnectionInfo()} method.
+ *
+ * @author Peter-Josef Meisch
+ */
+class ClusterConnection implements ExtensionContext.Store.CloseableResource {
```

Comment:
It would make sense to store both information, the endpoint and the handle to a cluster runner (Node client, Test containers) in the Jupiter context for separation of concerns.

Reply:
The ClusterConnection is stored in the ExtensionContext. Or do you mean having separate objects for the different connection types? I think, we will remove the client part when removing transport client stuff, so we then will only need the host/port/ssl part. Even when adding support for TestContainers (optional test depenedency?), the connection to the ES in a TestContainer will be defined by the URL

Reply:
See comment below in the actual extension.